packages/sampler/stories/moonstone-stories/VirtualGridList.js

```diff
@@ -51,16 +52,28 @@ const
 		);
 	};
 
-for (let i = 0; i < 1000; i++) {
+const updateDataSize = (dataSize) => {
 	const
-		count = ('00' + i).slice(-3),
-		text = `Item ${count}`,
-		subText = `SubItem ${count}`,
-		color = Math.floor((Math.random() * (0x1000000 - 0x101010)) + 0x101010).toString(16),
-		source = `http://placehold.it/300x300/${color}/ffffff&text=Image ${i}`;
+		itemNumberDigits = dataSize > 1 ? Math.floor(Math.log10(dataSize - 1)) + 1 : 0,
```

Comment:
How about (dataSize - 1 + '').length? In addition, we need to check if dataSize equals 0 or not.